compose/config/fields_schema.json

```diff
@@ -116,6 +116,7 @@
         "security_opt": {"type": "array", "items": {"type": "string"}, "uniqueItems": true},
         "stdin_open": {"type": "boolean"},
         "tty": {"type": "boolean"},
+        "ulimits": {"type": "object"},
```

Comment:
I think this is where most, or possibly all, the error checking should happen. Instead of just "object" you can specify the nested schema. There's a partial example of this with the environment key.